packages/web-components/src/components/va-pagination/va-pagination.tsx

```diff
@@ -233,102 +290,98 @@ export class VaPagination {
     i18next.on('languageChanged', () => {
       forceUpdate(this.el);
     });
+    this.resizeObserver.observe(this.el);
   }
 
   disconnectedCallback() {
     i18next.off('languageChanged');
+    this.resizeObserver.disconnect();
   }
 
-  render() {
-    const { ariaLabelSuffix, page, pages } = this;
+  /**
+   * =========================================
+   * Pagination render functions
+   * =========================================
+   */
+  private renderPreviousButton(
+    arrowClasses: string,
+    previousAriaLabel: string,
+  ): JSX.Element {
+    // Skip rendering if the viewport is too narrow to show Previous button
+    if (this.isMobileViewport) return null;
```

Comment:
I flipped the logic on these new UI render functions. If a condition exists that means this piece of pagination should not be rendered, I returned early using [guard clauses](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html). This helped me focus the return statements on the exact markup we wanted.